### PR TITLE
samples: central_bas: handle directed advertising

### PR DESCRIPTION
**Pull request by kapi-no**
The central_bas sample is now able to handle directed advertising.
Application connects to every device which uses it.

In Readme, users are recommended to test this sample with
peripheral_hids_mouse, so this change makes both applications more
compatible with each other.

This commit is based on the similar change done for the central_hids
sample: 4c87bfc83f61792c681ed50c5affabfb22266233.

Ref: NCSDK-4647

Signed-off-by: Kamil Piszczek <Kamil.Piszczek@nordicsemi.no>

### CODE REVIEW

**Comment by kapi-no** [comment]
Please take a look: @grochu, @KAGA164

File: samples/bluetooth/central_bas/prj.conf [hunk 1]
```diff
@@ -17,6 +17,7 @@ CONFIG_BT_GATT_BAS_C=y
 CONFIG_BT_SCAN=y
 CONFIG_BT_SCAN_FILTER_ENABLE=y
 CONFIG_BT_SCAN_UUID_CNT=1
+CONFIG_BT_PRIVACY=y
```
Comment (grochu) [inline samples/bluetooth/central_bas/prj.conf]: Is enabling this option required to get the directed advertising events?

Comment (kapi-no) [reply samples/bluetooth/central_bas/prj.conf]: Well, as far as I remember central devices get Non-Resolvable Private address from Zephyr BLE stack. With this address type, directed advertising is impossible. To make it compatible I need to use either fixed address using this Kconfig option:

CONFIG_BT_SCAN_WITH_IDENTITY

or Resolvable Private Address using this Kconfig option:

CONFIG_BT_PRIVACY

Please correct me @joerchan if this is no longer true.

Comment (KAGA164) [reply samples/bluetooth/central_bas/prj.conf]: You have to enable either `CONFIG_BT_SCAN_WITH_IDENTITY` or `CONFIG_BT_PRIVACY` to receive directed adv packet. `CONFIG_BT_PRIVACY` is recommended option.

Comment (grochu) [reply samples/bluetooth/central_bas/prj.conf]: Thanks for clarifying.

**Comment by kapi-no** [comment]
No doc changes required here.

I think this is ready to be merged.

**Pull request merged**